conjure-java-core/src/main/java/com/palantir/conjure/java/types/UnionGenerator.java

```diff
@@ -173,6 +174,40 @@ private static MethodSpec generateGetValue(ClassName baseClass) {
                     return builder.build();
                 })
                 .collect(Collectors.toList());
+        staticFactories.add(generateUnknownStaticFactory(unionClass, memberTypeDefs));
+        return staticFactories;
+    }
+
+    private static MethodSpec generateUnknownStaticFactory(ClassName unionClass, List<FieldDefinition> memberTypeDefs) {
+        String typeParam = "type";
+        String valueParam = "value";
+        MethodSpec.Builder builder = MethodSpec.methodBuilder("unknown")
+                .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+                .addParameter(String.class, typeParam)
+                .addParameter(Object.class, valueParam)
+                .returns(unionClass);
+        // begin switch statement
+        builder.beginControlFlow("switch($L)", Expressions.requireNonNull(typeParam, "Type is required"));
+        // add all cases
+        memberTypeDefs.forEach(memberTypeDef -> {
+            String memberName = sanitizeUnknown(memberTypeDef.getFieldName()).get();
+            builder.addCode("case $S:", memberName);
+            builder.addStatement(
+                    "throw new $T(\"$L: $L\")",
+                    SafeIllegalArgumentException.class,
+                    "Unknown type cannot be created as the provided type is known",
+                    memberName);
+        });
+        // add default case, which actually builds the unknown
+        builder.addCode("default:");
+        builder.addStatement(
+                "return new $T(new $T($L, $L))",
```

Comment:
nit: iirc `$N` represents name references
```suggestion
                "return new $T(new $T($N, $L))",
```